Source/Plugins/NetworkEvents/NetworkEvents.cpp

```diff
@@ -254,6 +323,14 @@ void NetworkEvents::process (AudioSampleBuffer& buffer)
         postTimestamppedStringToMidiBuffer (msg);
         networkMessagesQueue.pop();
     }
+
+    ScopedLock TTLlock(TTLqueueLock);
+    while (!TTLQueue.empty())
+    {
+        const StringTTL& TTLmsg = TTLQueue.front();
+        triggerEvent(TTLmsg);
+        TTLQueue.pop();
+    }
```

Comment:
Think this is fixed in newest commit. Is this what you mean by its own block?
`{
        ScopedLock TTLlock(TTLqueueLock);
        while (!TTLQueue.empty())
        {
            const StringTTL& TTLmsg = TTLQueue.front();
            triggerEvent(TTLmsg, timestamp);
            TTLQueue.pop();
        }
    }`

Reply:
Yup, the `TTLlock` part looks good; I was just saying you should enclose the part above with the `queueLock` in its own block as well, so that that lock is destroyed before we move on to processing the `TTLQueue`.

Reply:
Ahhh got it.